### PR TITLE
Add download button for full homeassistant log

### PR DESCRIPTION
**Pull request by mib1185**
## Proposed change

- adds a new download button to the settings log page
- renamed "Load Full Home Assistant Log" to "Show Full Home Assistant Log" to be more different to "Download Full Home Assistant Log"

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by spacegaier** [comment]
I like the idea. The only concern is that those buttons are really long (in German even longer since English is a comparatively short worded language).

One option might be to make a headline / collapsible section labeled "Full Home Assistant Log" and then below just have "Load Log" and "Download Log" buttons.

File: translations/frontend/en.json [hunk 1]
```diff
@@ -2561,7 +2561,8 @@
                         "info": "INFO",
                         "warning": "WARNING"
                     },
-                    "load_full_log": "Load Full Home Assistant Log",
+                    "load_full_log": "Show Full Home Assistant Log",
+                    "download_full_log": "Download Full Home Assistant Log",
```
Comment (spacegaier) [inline translations/frontend/en.json:2565]: Only change the `en.json` file in the `src/translations` folder. The rest will be updated automatically via the translation scripts.

**Review by spacegaier — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
There is a different proposal around, which I think we should run for, similar to the diagnostics download, which is filtered based on integration.

A general download button doesn't help much, as that can't be shared either.

**Comment by mib1185** [comment]
Would it be ok to add a warning, that the log could contain sensitiv data and user needs to care about to redact them before sharing the logs anyway?

File: translations/frontend/en.json (hunk 1 above)
Comment (mib1185) [reply translations/frontend/en.json]: thx for that hint

File: src/panels/config/logs/error-log-card.ts [hunk 2]
```diff
@@ -29,6 +30,9 @@ class ErrorLogCard extends LitElement {
               <mwc-button raised @click=${this._refreshErrorLog}>
                 ${this.hass.localize("ui.panel.config.logs.load_full_log")}
               </mwc-button>
+              <mwc-button raised @click=${this._downloadLogs}>
```
Comment (kubawolanin) [inline src/panels/config/logs/error-log-card.ts:33]:
```suggestion
              <mwc-button @click=${this._downloadLogs}>
```

image

**Review by kubawolanin — CHANGES_REQUESTED** [review]

**Comment by bramkragten** [comment]
I don't think we should add this, but if we do, I think it is better to do it inside `error-log-card`, like the refresh button that is in there now.

**Pull request closed without merging**